tests/components/atag/test_climate.py

```diff
@@ -93,18 +86,18 @@ async def test_incorrect_modes(
         assert hass.states.get(CLIMATE_ID).state == STATE_UNKNOWN
 
 
-async def test_update_service(
-    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+async def test_update_failed(
+    hass: HomeAssistant,
+    aioclient_mock: AiohttpClientMocker,
 ) -> None:
-    """Test the updater service is called."""
-    await init_integration(hass, aioclient_mock)
+    """Test data is not destroyed on update failure."""
+    entry = await init_integration(hass, aioclient_mock)
     await async_setup_component(hass, HA_DOMAIN, {})
-    with patch("pyatag.AtagOne.update") as updater:
-        await hass.services.async_call(
-            HA_DOMAIN,
-            SERVICE_UPDATE_ENTITY,
-            {ATTR_ENTITY_ID: CLIMATE_ID},
-            blocking=True,
-        )
+    assert hass.states.get(CLIMATE_ID).state == HVAC_MODE_HEAT
+    coordinator = hass.data[DOMAIN][entry.entry_id]
```

Comment:
We don't want to access integration details in the test. Patch time and move time forward instead to have the coordinator update. We can use this test helper:

https://github.com/home-assistant/core/blob/514516baccd8fe6a1a566db633247c2eaef51874/tests/common.py#L361-L363